action_server/src/sema4ai/action_server/_ask_user.py

```diff
@@ -0,0 +1,106 @@
+import typing
+from logging import getLogger
+
+from sema4ai.action_server.package._package_publish import Organization
+
+log = getLogger(__name__)
+
+
+def ask_user_input_to_proceed(msg: str) -> bool:
+    """
+    Args:
+        msg: The message to query user (should end with something as: "(y/n)\n").
+
+    Returns:
+        True to proceed and False otherwise.
+    """
+    try:
+        while (c := input(msg).lower().strip()) not in ("y", "n"):
+            continue
+        if c == "n":
+            return False
+        # otherwise 'y' keep on going...
+    except KeyboardInterrupt:
+        return False
+
+    return True
+
+
+def ask_user_to_provide_access_credentials() -> str:
+    """
+    Returns:
+        Control Room access credentials in valid form.
+    """
+    from sema4ai.action_server.vendored_deps.termcolors import bold_red
+
+    while True:
+        try:
+            access_credentials = str(input("Enter Control Room access credentials: "))
+            if access_credentials:
+                _, _ = access_credentials.split(":")  # validate credentials form
+                return access_credentials
+            else:
+                log.error(
+                    bold_red(
+                        "Invalid credentials, please try again with form: <keyid:secret>."
+                    )
+                )
+        except ValueError:
+            log.error(
+                bold_red(
+                    "Invalid access credentials, please try again with form: <keyid:secret>."
+                )
+            )
+
+
+def ask_user_to_choose_organization(
+    organizations: typing.List[Organization],
+) -> str:
+    """
+    Args:
+        organizations: List of organizations with name and ID.
+
+    Returns:
+        Organization ID.
+    """
+    from sema4ai.action_server.vendored_deps.termcolors import bold_red, colored
+
+    for index, name in enumerate(
+        [organization.name for organization in organizations], start=1
+    ):
+        log.info(colored(f" > {index}. {name}", "cyan"))
+
+    while True:
+        try:
+            choise = int(input("Enter the number of your choice: "))
+            if 1 <= choise <= len(organizations):
+                return organizations[choise - 1].id
+            else:
+                log.error(bold_red("Invalid number, please try again."))
+        except ValueError:
+            log.error(bold_red("Invalid input, please enter a number."))
+
+
+def ask_user_for_hostname() -> str:
+    """
+    Returns:
+        Control Room hostname.
+    """
+    from urllib.parse import urlparse
+
+    from sema4ai.action_server._keyring import DEFAULT_CR_HOSTNAME
+    from sema4ai.action_server.vendored_deps.termcolors import bold_red
+
+    while True:
+        try:
+            hostname = input(f"Enter Control Room hostname ({DEFAULT_CR_HOSTNAME}): ")
```

Comment:
Question: instead of `DEFAULT_CR_HOSTNAME`, shouldn't the default be what the user entered the last time (it seems we're saving that, right?)

Or maybe it should show an option with something as default/last/specify new?

Reply:
We are storing it. I think it makes sense to show what the user set last time, if the user wants to just change the access credentials. I'll change it.